### PR TITLE
iox-#1812 Fix `expected::operator bool` doxygen

### LINKED ISSUE #1812: `iox::cxx::expected::operator bool` has wrong doxygen comment
## Required information

**Operating system:**
All OSs

**Compiler version:**
All compilers

**Observed result or behaviour:**
Look here: https://github.com/eclipse-iceoryx/iceoryx/blob/master/iceoryx_hoofs/include/iceoryx_hoofs/cxx/expected.hpp#L239
The `operator bool` doxygen comment is actually completely opposite of its behaviour!

**Expected result or behaviour:**
The comment should specify that `operator bool` actually returns true if the `expected` instance signals success.

**Conditions where it occurred / Performed steps:**

### PR DESCRIPTION
**Pull request by JakubSosnovec**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [ ] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Branch follows the naming format (`iox-123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [ ] All checks have passed (except `task-list-completed`)
1. [ ] All touched (C/C++) source code files from `iceoryx_hoofs` are added to `./clang-tidy-diff-scans.txt`
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [ ] Copyright owner are updated in the changed files
- [x] All touched (C/C++) source code files from `iceoryx_hoofs` have been added to `./clang-tidy-diff-scans.txt`
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [ ] All open points are addressed and tracked via issues

## References

- Closes #1812

### CODE REVIEW

**Comment by JakubSosnovec** [comment]
Not sure whether this even deserves a mention in the Changelog...

File: iceoryx_hoofs/include/iceoryx_hoofs/cxx/expected.hpp [hunk 1]
```diff
@@ -234,8 +234,8 @@ class IOX_NO_DISCARD expected<ErrorType> : public FunctionalInterface<expected<E
     template <typename... Targs>
     static expected create_error(Targs&&... args) noexcept;
 
-    /// @brief  returns true if the expected contains an error otherwise false
-    /// @return bool which contains true if the expected contains an error
+    /// @brief  returns true if the expected signals success otherwise false
+    /// @return bool which contains true if the expected signals success
```
Comment (FerdinandSpitzschnueffler) [inline iceoryx_hoofs/include/iceoryx_hoofs/cxx/expected.hpp:238]: Could you please change the comment in line 355 as well?

Comment (JakubSosnovec) [reply iceoryx_hoofs/include/iceoryx_hoofs/cxx/expected.hpp:238]: Ah, missed that one. Changed now.

**Review by FerdinandSpitzschnueffler — APPROVED** [review]

**Comment by JakubSosnovec** [comment]
> @JakubSosnovec oh no, this is already fixed in https://github.com/eclipse-iceoryx/iceoryx/pull/1728 which is not yet merged. I think this is the second time this happens for your 😅

Oh damnit! :D Indeed, thats the second time. Next time, I will first check all pending PRs before pushing anything!

Alright, closing in favor of #1728 .

**Comment by elBoberido** [comment]
@JakubSosnovec just some bad timing and it took quite long to merge the PR. Alternatively to checking the PRs, you could also use gitter and ask if there is already someone working on your finding :)

**Comment by JakubSosnovec** [comment]
> @JakubSosnovec just some bad timing and it took quite long to merge the PR. Alternatively to checking the PRs, you could also use gitter and ask if there is already someone working on your finding :)

Oh no worries, I am not using the master branch anyway, just wanted to have this fixed for the future :)

**Pull request closed without merging**